src/context.ml

```diff
@@ -23,6 +23,20 @@ module Env_nodes = struct
     { context: Dune_env.Stanza.t option
     ; workspace: Dune_env.Stanza.t option
     }
+
+  let extra_vars ~profile env_nodes =
+    let make_map = function
+      | None ->
+        Env.Map.empty
+      | Some stanza ->
+        match Dune_env.Stanza.find stanza ~profile with
+        | None -> Env.Map.empty
+        | Some {env; _} -> Env.Map.of_list_exn env
```

Comment:
This is more a matter of taste, but I prefer to write this code monadically, and only get `Option.value ~default:Env.Map.empty` at the end. So something like:

```
let make_map stanza =
  Option.value ~default:Env.Map.empty (
    stanza >>= Dune_env.Stanza.find stanza ~profile >>| fun {env; _} -> Env.Map.of_list_exn env
  )
```